llvm/lib/Target/AIE/AIE2InstrInfo.cpp

```diff
@@ -829,7 +832,11 @@ bool AIE2InstrInfo::expandPostRAPseudo(MachineInstr &MI) const {
   case AIE2::PseudoMove: {
     Register Dst = MI.getOperand(0).getReg();
     Register Src = MI.getOperand(1).getReg();
-    BuildMI(MBB, MI, DL, get(AIE2::MOV_mv_scl), Dst)
+    const unsigned MOVSclOpcode =
+        (AIE2::eRRegClass.contains(Dst) && AIE2::eRRegClass.contains(Src))
+            ? AIE2::MOV_OR_GPR_pseudo
+            : AIE2::MOV_mv_scl;
+    BuildMI(MBB, MI, DL, get(MOVSclOpcode), Dst)
```

Comment:
This looks very similar to what we do in copyPhysReg. Can we reuse?

Reply:
Sure will abstract it out!